interface/patient_file/summary/demographics_full.php

```diff
@@ -797,6 +811,56 @@ function end_group() {
     $("select").change(function() {
         checkSkipConditions();
     });
+
+//This code deals with demographics before save action -
+<?php if ($hook):?>
+
+	var f = $("form");
+
+	// Use hook to open the controller and get the new patient validation .
+	// when no params are sent this window will be closed from the zend controller.
+	var url ='<?php echo  $GLOBALS['web_root']."/interface/modules/zend_modules/public/patientvalidation";?>';
+	$("#submit").attr("type","button");
+	$("#submit").attr("name","btnSubmit");
+	$("#submit").attr("id","btnSubmit");
+	$("#btnSubmit").click(function( event ) {
+		if(!submitme(<?php echo $GLOBALS['new_validate'] ? 1 : 0;?>,event,'DEM',constraints)){
+			event.preventDefault();
+			return;
+		}
+		somethingChanged = false;
```

Comment:
This will bypass Rod's mechanism to avoid losing data if screen times out or user leaves screen. However, I am guessing you need to do this prior to showing the Zend popup to avoid the "You have unsaved changes." alert error. Correct? If so, this is fine, since assume the Zend popup will save the patient data after process it.